skytemple_files/hardcoded/guest_pokemon.py

```diff
@@ -0,0 +1,269 @@
+        # True if Hidden Land-like restrictions should be enabled in this dungeon after it's cleared
+        self.hlr_cleared = hlr_cleared
+
+    def __eq__(self, other):
+        if not isinstance(other, ExtraDungeonDataEntry):
+            return False
+        else:
+            return self.guest1_index == other.guest1_index and \
+                self.guest2_index == other.guest2_index and \
+                self.hlr_uncleared == other.hlr_uncleared and \
+                self.disable_recruit == other.disable_recruit and \
+                self.side01_check == other.side01_check and \
+                self.hlr_cleared == other.hlr_cleared
+
+    def to_bytes(self) -> bytes:
+        buffer = bytearray(2)
+
+        entry = 0
+        if self.guest1_index != -1:
+            entry = self.guest1_index + 1
+        if self.hlr_uncleared:
+            entry |= 0x80
+        if self.disable_recruit:
+            entry |= 0x40
+        if self.guest2_index != -1:
+            entry += self.guest2_index + 1 << 8
+        if self.side01_check:
+            entry |= 0x8000
+        if self.hlr_cleared:
+            entry |= 0x4000
+
+        write_uintle(buffer, entry, 0, 2)
+        return buffer
+
+    @classmethod
+    def from_bytes(cls, b: bytes) -> 'ExtraDungeonDataEntry':
+        data_int = read_uintle(b, 0, 2)
+        guest1_index = (data_int & 0x3F) - 1
+        guest2_index = ((data_int & 0x3F00) >> 8) - 1
+        return cls(guest1_index, guest2_index, bool(data_int & 0x80), bool(data_int & 0x40), bool(data_int & 0x8000),
+                   bool(data_int & 0x4000))
+
+    @classmethod
+    def _valid_index(cls, index: int):
+        """
+        Returns true if a given guest pokémon index is valid.
+        Valid values are indexes 0 to 62 and -1, which is used as an empty value
+        """
+        return -1 <= index <= 62
+
+
+class GuestPokemon(AutoString):
+    def __init__(self, unk1: int, poke_id: int, unk2: int, moves: List[int], hp: int, level: int,
+                 iq: int, atk: int, sp_atk: int, def_: int, sp_def: int, unk3: int, exp: int):
+        self.unk1 = unk1
+        self.poke_id = poke_id
+        self.unk2 = unk2
+        self.moves = moves
+        self.hp = hp
+        self.level = level
+        self.iq = iq
+        self.atk = atk
+        self.sp_atk = sp_atk
+        self.def_ = def_
+        self.sp_def = sp_def
+        self.unk3 = unk3
+        self.exp = exp
+
+    def __eq__(self, other):
+        if not isinstance(other, GuestPokemon):
+            return False
+        else:
+            return self.unk1 == other.unk1 and \
+                   self.poke_id == other.poke_id and \
+                   self.unk2 == other.unk2 and \
+                   self.moves == other.moves and \
+                   self.hp == other.hp and \
+                   self.level == other.level and \
+                   self.iq == other.iq and \
+                   self.atk == other.atk and \
```

Comment:
Special Attack missing (also in is_null)

Reply:
Fixed, thanks!